### PR TITLE
Fixed #25032 - When next value is missing, redirect to main /admin/ page, not to self.

### PR DESCRIPTION
**Pull request by adelton**

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Makes sense to me. Could you add a test (probably in `tests/admin_views`)?

**Comment by adelton** [comment]
> Makes sense to me. Could you add a test (probably in tests/admin_views)?

Added in 46e91f248e62c7754275b906be27ce665850cbbf.

I went with just matching the hidden field in the response because I saw it done that way in other tests. Ideally we'd want to check the semantics (exact forms, extact fields, not the exact syntax) or behaviour but I did not find any utility methods to submit the form exactly as received (with login/password filled).

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -1480,6 +1481,14 @@ def test_login(self):
         form = login.context[0].get('form')
         self.assertEqual(form.errors['username'][0], 'This field is required.')
 
+        # Test logon redirects when going directly to /admin/login (#25032)
+        self.client.get(reverse('admin:logout'))
+        response = self.client.get(reverse('admin:login'), follow=False)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, '<input type="hidden" name="next" value="%s" />' % reverse('admin:index'))
```
Comment (timgraham) [inline tests/admin_views/tests.py]: I'd suggest to check `response.context[REDIRECT_FIELD_NAME]` instead.

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -1480,6 +1481,14 @@ def test_login(self):
...
+        self.assertNotContains(response, '<input type="hidden" name="next" value="%s" />' % reverse('admin:login'))
+        self.client.get(reverse('admin:logout'))
```
Comment (timgraham) [inline tests/admin_views/tests.py]: there's no need to "cleanup" by logging out as each test creates a new test client

File: tests/admin_views/tests.py [hunk 3]
```diff
@@ -1480,6 +1481,14 @@ def test_login(self):
         form = login.context[0].get('form')
         self.assertEqual(form.errors['username'][0], 'This field is required.')
 
+        # Test logon redirects when going directly to /admin/login (#25032)
+        self.client.get(reverse('admin:logout'))
```
Comment (timgraham) [inline tests/admin_views/tests.py]: a separate test method would be preferred

**Comment by timgraham** [comment]
You can squash commits when updating, thanks.

**Comment by adelton** [comment]
Updated per your suggestions, rebased on master, squashed -> a9e93ad27e8a1928937118a4291b39f6d84c66da.

**Comment by timgraham** [comment]
merged in 335368410210ce0e27e0068b3a2a6cdf4baa956b, thanks!

**Pull request closed without merging**